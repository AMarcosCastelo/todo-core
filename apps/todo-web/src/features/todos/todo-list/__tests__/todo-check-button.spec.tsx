import { vi } from 'vitest';

import { baseRender } from '~/utils/testing';

import TodoCheckButton from '../todo-check-button';

describe('TodoCheckButton', () => {
  const defaultProps = {
    handleClick: vi.fn(),
  };
  const component = (customProps = {}) => (
    <TodoCheckButton {...customProps} {...defaultProps} />
  );

  it('should render correctly', () => {
    const screen = baseRender(component());

    expect(screen.container).toMatchSnapshot();
  });

  it('should render the icon', () => {
    const screen = baseRender(component());

    expect(screen.getByTestId('check-icon')).toBeInTheDocument();
  });
});
