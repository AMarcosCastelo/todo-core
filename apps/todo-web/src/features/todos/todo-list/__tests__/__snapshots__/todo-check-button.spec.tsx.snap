// Vitest Snapshot v1

exports[`TodoCheckButton > should render correctly 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <div
      class="nextui-c-ftJsHY nextui-tooltip-button"
      role="button"
      tabindex="-1"
    >
      <button
        aria-label="Check button"
        class="nextui-c-eWzLKI"
      >
        <svg
          class="lucide lucide-check"
          data-testid="check-icon"
          fill="none"
          height="20"
          stroke="var(--nextui-colors-blue600)"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
